ksa_compliance/standard_doctypes/sales_invoice.py

```diff
@@ -101,3 +101,11 @@ def should_enable_zatca_for_invoice(invoice_id: str) -> bool:
 
     posting_date = frappe.db.get_value('Sales Invoice', invoice_id, 'posting_date')
     return posting_date >= start_date
+
+
+def calculate_tax_amount(self):
+    if self.items:
+        for item in self.items:
+            item_tax_rate = frappe.get_value("Item Tax Template Detail", {"parent": item.item_tax_template}, ['tax_rate'])
+            item.custom_tax_total = (item.amount * item_tax_rate) / 100
```

Comment:
These calculations do not match the ones we do in JavaScript in `ksa_compliance/public/js/sales_invoice_item.js`. We're using the item rate there, not the amount, so in JavaScript it calculates the taxes per item (not per line, since it doesn't include quantity). We should either update the JavaScript to match this or remove it entirely and rely on the pre-save hook to fill it in.